holoviews/plotting/plot.py

```diff
@@ -1457,8 +1457,16 @@ def _get_frame(self, key):
 
         key_map = dict(zip([d.name for d in self.dimensions], key))
         for path, item in self.layout.items():
-            frame = item.map(lambda x: get_plot_frame(x, key_map, cached=cached),
-                             ['DynamicMap', 'HoloMap'])
+            clone = item.map(lambda x: x)
+
+            # Ensure that DynamicMaps in the cloned frame have
+            # identical callback inputs to allow memoization to work
+            for it1, it2 in zip(item.traverse(lambda x: x), clone.traverse(lambda x: x)):
+                if isinstance(it1, DynamicMap):
+                    with util.disable_constant(it2.callback):
+                        it2.callback.inputs = it1.callback.inputs
+            frame = clone.map(lambda x: get_plot_frame(x, key_map, cached=cached),
+                              [DynamicMap, HoloMap], clone=False)
```

Comment:
This stuff is a bit ugly and should probably become a utility. The main problem is that ``.map`` is being abused a bit.